### PR TITLE
Implement ws close reason code

### PR DESCRIPTION
**Pull request by ad31c0**
Implement reason code when closing ws connection.
Always close the STT session when a NoMatch is received.

### CODE REVIEW

**Review by gcampax — COMMENTED** [review]
Can we add a comment explaining what those error codes mean?

**Comment by ad31c0** [comment]
Sure, added an in-line comment about ws error codes used.

Implemented also the capability to change voice gender for TTS.

File: nlp/voice/index.js [hunk 1]
```diff
@@ -180,6 +194,8 @@ router.post('/:locale/voice/tts', iv.validatePOST({ text: 'string' }, { json: tr
 router.post('/@:model_tag/:locale/voice/stt', upload.single('audio'), restSTT);
 router.post('/@:model_tag/:locale/voice/query', upload.single('audio'),
     iv.validatePOST({ metadata: 'string' }, { json: true }), restSTTAndNLU);
-router.post('/@:model_tag/:locale/voice/tts', iv.validatePOST({ text: 'string' }, { json: true }), tts);
+
+// provide TTS API keyed off to :gender, so people can choose the voice gender to be used
+router.post('/@:gender/:locale/voice/tts', iv.validatePOST({ text: 'string' }, { json: true }), tts);
```
Comment (gcampax) [inline nlp/voice/index.js:199]: No. The previous API was correct: the @-sign denotes a model tag. The reason for this API design is that the whole "https://nlp.almond.stanford.edu/@com.foo" can be specified as a NLP server URL in Genie, so all API endpoints that are available at the root must be available also under a specific model tag (even if the model tag doesn't affect the model at all, as in the case of STT/TTS).
Use the body parameters to specify a gender.

**Review by gcampax — CHANGES_REQUESTED** [review]
The gender changes are not related to the error code changes. Please make two separate commits (it's ok if they are in the same PR).

File: nlp/voice/index.js (hunk 1 above)
Comment (ad31c0) [reply nlp/voice/index.js]: ok, now I understand why those are there even if unused.

**Review by gcampax — APPROVED** [review]
Ok

**Pull request merged**